src/vmm/src/guest_config/common/mod.rs

```diff
@@ -0,0 +1,158 @@
+#[cfg(target_arch = "aarch64")]
+#[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize)]
+#[allow(unused)]
+pub struct Aarch64RegisterMap {
+    register_map: HashMap<RegisterPointer<u64>, RegisterValue<u128>>,
+}
+
+/// A trait type to be customized for each CPU architecture
+/// to modify CPU configuration for the Firecracker guest
+pub trait ConfigurationModifier {
+    /// Applies the CPUTemplate against the CPUConfiguration
+    fn apply_template(&self, cpu_template: CustomCpuTemplate) -> Box<CpuConfiguration>;
+}
+
+/// CPU configuration that can be of the host or the guest.
+/// Expected to be populated in part or modified by
+/// types implementing ConfigurationModifier
+#[allow(unused)]
+pub struct CpuConfiguration {
+    /// Hashed set containing register configuration
+    /// divided up by type of register. See `RegisterSet`
+    pub register_config: Vec<RegisterSet>,
+}
+
+impl CpuConfiguration {
+    /// Factory method to instantiate CPUConfiguration
+    /// with the provided `RegisterSet`
+    pub fn new(register_config: Vec<RegisterSet>) -> CpuConfiguration {
+        CpuConfiguration { register_config }
+    }
+}
+
+/// Template indicating a modification of CPU configuration
+/// to be applied against the `CPUConfiguration` type
+#[allow(unused)]
+#[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize)]
+pub struct CustomCpuTemplate {
+    /// Modifier masks with pointer data for the modifiers
+    /// intended registers
+    pub modifiers: Vec<RegisterModifierSet>,
+}
+
+/// Modifier for a CPU configuration containing
+/// the location of the value to be modified,
+/// and provides a value to be applied alongside
+/// a filter to be applied
+#[allow(unused)]
+#[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize)]
+pub struct RegisterModifier<T: Numeric> {
+    /// Location of the register to be modified
+    /// by the mask value and filter
+    pub pointer: RegisterPointer<T>,
+    /// Mask value to be applied
+    pub value: T,
+    /// Mask value is to be applied according
+    /// to what is allowed through by the filter
+    pub filter: T,
+}
+
+/// Numeric wrapper type acting as a marker type
+/// for the register's pointer
+#[allow(unused)]
+#[derive(Clone, Debug, Deserialize, Eq, Hash, PartialEq, Serialize)]
+pub enum RegisterPointer<T: Numeric> {
+    /// Pointer(location) of a specific register
+    NumericPointer(T),
+    /// Register pointer for CPUID that is Leaf/Subleaf aware
+    CpuidPointer(CpuidRegisterPointer),
+}
+
+/// Numeric wrapper type acting as a marker type
+/// for the register's value
+#[allow(unused)]
+#[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize)]
+pub struct RegisterValue<T: Numeric> {
+    /// Value of a specific register
+    pub value: T,
+}
+
+/// CPUID register enumeration
+#[allow(unused)]
+#[allow(missing_docs)]
+#[derive(Clone, Debug, Deserialize, Eq, Hash, PartialEq, Serialize)]
+pub enum CpuidRegister {
+    Eax,
+    Ebx,
+    Ecx,
+    Edx,
+}
```

Comment:
Does this need to be a common type? Isn't it only used for x86?

Reply:
It is also used for the API (or any other potential interface in the future like CLI).